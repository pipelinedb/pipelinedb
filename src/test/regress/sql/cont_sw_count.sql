CREATE CONTINUOUS VIEW test_count AS SELECT k::text, COUNT(*) FROM cqswcount_stream WHERE arrival_timestamp > clock_timestamp() - interval '60 second' GROUP BY k;

INSERT INTO cqswcount_stream (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x');
INSERT INTO cqswcount_stream (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x'), ('y'), ('y'), ('y'), ('y'), ('y'), ('y');

SELECT * FROM test_count ORDER BY k;

SELECT pg_sleep(1);

INSERT INTO cqswcount_stream (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x');
INSERT INTO cqswcount_stream (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x'), ('y'), ('y'), ('y'), ('y'), ('y'), ('y');

SELECT * FROM test_count ORDER BY k;

DROP CONTINUOUS VIEW test_count;

-- Truncate the timestamp to second so that we don't create 6 second buckets (step_factor = 10)
CREATE CONTINUOUS VIEW sw_count0 AS SELECT COUNT(*) FROM cqswcount_stream WHERE second(arrival_timestamp) > clock_timestamp() - interval '60 second';

CREATE VIEW sw_count1 AS SELECT combine(count) FROM sw_count0_mrel0 WHERE arrival_timestamp > clock_timestamp() - interval '5 second';

INSERT INTO cqswcount_stream (k) VALUES ('x'), ('x');

SELECT * FROM sw_count0;
SELECT * FROM sw_count1;

SELECT pg_sleep(6);

SELECT * FROM sw_count0;
SELECT * FROM sw_count1;
