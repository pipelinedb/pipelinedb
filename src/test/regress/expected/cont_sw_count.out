CREATE CONTINUOUS VIEW test_count AS SELECT k::text, COUNT(*) FROM cqswcount_stream WHERE arrival_timestamp > clock_timestamp() - interval '60 second' GROUP BY k;
INSERT INTO cqswcount_stream (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x');
INSERT INTO cqswcount_stream (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x'), ('y'), ('y'), ('y'), ('y'), ('y'), ('y');
SELECT * FROM test_count ORDER BY k;
 k | count 
---+-------
 x |    12
 y |     6
(2 rows)

SELECT pg_sleep(1);
 pg_sleep 
----------
 
(1 row)

INSERT INTO cqswcount_stream (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x');
INSERT INTO cqswcount_stream (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x'), ('y'), ('y'), ('y'), ('y'), ('y'), ('y');
SELECT * FROM test_count ORDER BY k;
 k | count 
---+-------
 x |    24
 y |    12
(2 rows)

DROP CONTINUOUS VIEW test_count;
-- Truncate the timestamp to second so that we don't create 6 second buckets (step_factor = 10)
CREATE CONTINUOUS VIEW sw_count0 AS SELECT COUNT(*) FROM cqswcount_stream WHERE second(arrival_timestamp) > clock_timestamp() - interval '60 second';
CREATE VIEW sw_count1 AS SELECT combine(count) FROM sw_count0_mrel0 WHERE arrival_timestamp > clock_timestamp() - interval '5 second';
INSERT INTO cqswcount_stream (k) VALUES ('x'), ('x');
SELECT * FROM sw_count0;
 count 
-------
     2
(1 row)

SELECT * FROM sw_count1;
 combine 
---------
       2
(1 row)

SELECT pg_sleep(6);
 pg_sleep 
----------
 
(1 row)

SELECT * FROM sw_count0;
 count 
-------
     2
(1 row)

SELECT * FROM sw_count1;
 combine 
---------
       0
(1 row)

